realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -138,6 +143,279 @@ public boolean load() {
     }
 
     /**
+     * Sets the value to {@code null} for the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @throws IllegalArgumentException if field name doesn't exist or isn't nullable.
+     * @throws RealmException if the field is a {@link io.realm.annotations.PrimaryKey} field.
+     */
+    public void setNull(@Nonnull String fieldName) {
+        verifyString(fieldName);
+        realm.checkIfValid();
+        collection.setNull(getColumnIndex(fieldName));
+    }
+
+
+    /**
+     * Sets the {@code boolean} value of the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't boolean.
+     */
+    public void setBoolean(@Nonnull String fieldName, @Nullable Boolean value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
```

Comment:
Yes.  Totally agree that this is an essential question.  Please see https://github.com/realm/realm-java/pull/5133#discussion_r134476731 for my reasoning for this choice.  We need to just take a vote or something.  I can see either primitive params and an explicit `setNull` or nullable boxed params.  Either one makes sense.